krm-functions/lib/fn-runtime/design.md

```diff
@@ -0,0 +1,44 @@
+# fn-runtime
+
+The fn-runtime Project is a set of go libraries for building Kpt functions/controllers that are part of a "`Condition` choreography" also called a "`Condition` dance.
+
+The "`Condition` choreography/dance" is a set of kpt function and controllers that perform independent actions on a kpt package. Together and in harmony a declaritive outcome is achieved on a kpt package (a set of KRM resources).
+
+When comparing the conditional dance with a regular kubernetes controller built upon the controller runtime, we can make a number of analogies.
+- The event driven nature on a For object in the controller runtime is taken by porch, which periodically present the functions/controllers the latest status of the package. The fn/controllers provide an indepempodent operation on the package based on the resource presented.
```

Comment:
s/porch/Porch/
s/present/presents/
s/the latest/with the latest/
s/indepempotent/idempotent/

Reply:
fixed